sdk/search/azure-search/src/main/java/com/azure/search/models/DataSource.java

```diff
@@ -0,0 +1,234 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+// Code generated by Microsoft (R) AutoRest Code Generator.
+// Changes may cause incorrect behavior and will be lost if the code is
+// regenerated.
+
+package com.azure.search.models;
+
+import com.azure.core.annotation.Fluent;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+/**
+ * Represents a datasource definition, which can be used to configure an
+ * indexer.
+ */
+@Fluent
+public final class DataSource {
```

Comment:
We're missing the subtype property that is included on the backend DataSource definition.  Its not discussed in the documentation though so maybe its left out intentionally, just mentioning it.

Reply:
@careyjmac The backend definitions are not really useful for reviewing client-side code, since we do things for the sake of implementation convenience that never show up on the wire. The Swagger spec should be the source of truth for what shows up in the generated code.